### PR TITLE
Script to generate ERD

### PR DESCRIPTION
**Pull request by dacook**
#### What? Why?

A new shell script to easily and consistently generate the latest Entity Relationship Diagram based on the rails application models. 
This is a useful reference for developing the application, APIs and SQL queries. 
Some people may need to refer to it without wanting to install and generate it themselves, so the pdf is saved for easy access at this URL (once merged): https://github.com/openfoodfoundation/openfoodnetwork/blob/master/doc/erd.pdf

It would be cool to automatically generate and publish this in CI, but I didn't attempt that here.

Discussion: https://openfoodnetwork.slack.com/archives/C2GQ45KNU/p1663121858195469

#### What should we test?
After `bundle install`,
Execute this script: `script/erd`

#### Release notes

Data model diagram (ERD) stored and available at https://github.com/openfoodfoundation/openfoodnetwork/blob/master/doc/erd.pdf 
__link to be verified before publishing__

Changelog Category: Technical changes

#### Documentation updates
A link to the ERD can now be provided from the wiki, perhaps under the section **Data & APIs**. Rather than store the pdf in the wiki, I recommend linking to the codebase, so that users always see the most recent diagram.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -812,4 +821,4 @@ RUBY VERSION
    ruby 3.0.3p157
 
 BUNDLED WITH
-   2.1.4
+   2.2.32
```
Comment (jibees) [inline Gemfile.lock:824]: I don't think this change is necessary/wanted.

**Review by jibees — CHANGES_REQUESTED** [review]
After testing, one thing I noticed ... 
The bundle version has changed, I don't think it's that we want.

File: Gemfile.lock (hunk 1 above)
Comment (dacook) [reply Gemfile.lock]: Oops, good pickup thanks! Updated the commit.

**Comment by dacook** [comment]
Thanks @jibees , I forgot to `install-bundler` first. I've added a fix to make that script work on macOS by default too.

**Comment by dacook** [comment]
PS once merged, would it be ok for me to add a new page to the wiki under _Data & APIs_?
Or is there another process for updating the wiki?

**Review by jibees — APPROVED** [review]

**Comment by jibees** [comment]
Testing notes:

Once I ran `bundle`, I had this post install message:
```
Post-install message from ruby-graphviz:

You need to install GraphViz (https://graphviz.org) to use this Gem.

For more information about Ruby-Graphviz :
* Doc: https://rdoc.info/github/glejeune/Ruby-Graphviz
* Sources: https://github.com/glejeune/Ruby-Graphviz
* Issues: https://github.com/glejeune/Ruby-Graphviz/issues
```

Don't know if it's relevant. 

Then I ran `script/erd` and I got this file under `doc/` folder.
[erd.pdf](https://github.com/openfoodfoundation/openfoodnetwork/files/9613878/erd.pdf)


Good to go then! 👍 🎉 

> PS once merged, would it be ok for me to add a new page to the wiki under Data & APIs?

Yes, I guess it's a good idea!

**Comment by sigmundpetersen** [comment]
Can we merge this @jibees ? 🙂

**Comment by sigmundpetersen** [comment]
@dacook what do think of the `ruby-graphviz` message?
Anything that needs to be done in a follow-up PR?

**Comment by dacook** [comment]
@sigmundpetersen , I think JB is right: if it works, then it we don't need to worry :)

This is actually a system dependency that can't be managed by Bundler anyway. It's only (maybe) required for a developer to run the script, not required on CI or production. So there's no code change required.

The post-install message is clear, but I will add a note to the new wiki page just in case it helps someone.

**Comment by dacook** [comment]
I've added a new page under _Data & APIs_ in the wiki: https://github.com/openfoodfoundation/openfoodnetwork/wiki/Data-model-diagram-%28ERD%29

Let me know if you have any suggestions or feel free to update it.

**Pull request merged**